reconcile/terraform_tgw_attachments.py

```diff
@@ -1,157 +1,213 @@
+    # to get the information from OCM. it likely means that
+    # there is no OCM at all.
+    account["assume_role"] = (
+        provided_assume_role
+        if provided_assume_role
+        else ocm.get_aws_infrastructure_access_terraform_assume_role(
+            cluster_name, account["uid"], account["terraformUsername"]
+        )
+    )
+    account["assume_region"] = region
+    account["assume_cidr"] = cidr_block
+    return account
+
+
+def _build_accepter(
+    peer_connection: dict,
+    account: dict,
+    region: str,
+    cidr_block: str,
+    awsapi: AWSApi,
+) -> dict:
+    (vpc_id, route_table_ids, subnets_id_az) = awsapi.get_cluster_vpc_details(
+        account,
+        route_tables=peer_connection.get("manageRoutes"),
+        subnets=True,
+    )
+    return {
+        "cidr_block": cidr_block,
+        "region": region,
+        "vpc_id": vpc_id,
+        "route_table_ids": route_table_ids,
+        "subnets_id_az": subnets_id_az,
+        "account": account,
+    }
+
+
+def _build_requester(
+    peer_connection: dict,
+    account: dict,
+    tgw: dict,
+) -> dict:
+    return {
+        "tgw_id": tgw["tgw_id"],
+        "tgw_arn": tgw["tgw_arn"],
+        "region": tgw["region"],
+        "routes": tgw.get("routes"),
+        "rules": tgw.get("rules"),
+        "hostedzones": tgw.get("hostedzones"),
+        "cidr_block": peer_connection.get("cidrBlock"),
+        "account": account,
+    }
+
+
+def _build_ocm_map(
+    clusters: list,
+    settings: Optional[Mapping[str, Any]],
 ):
-    settings = queries.get_secret_reader_settings()
-    clusters = queries.get_clusters_with_peering_settings()
     with_ocm = any(c.get("ocm") for c in clusters)
-    if with_ocm:
-        ocm_map = OCMMap(
-            clusters=clusters, integration=QONTRACT_INTEGRATION, settings=settings
-        )
-    else:
+    return (
+        OCMMap(clusters=clusters, integration=QONTRACT_INTEGRATION, settings=settings)
+        if with_ocm
         # this is a case for an OCP cluster which is not provisioned
         # through OCM. it is expected that an 'assume_role' is provided
-        # on the tgw defition in the cluster file.
-        ocm_map = {}
-
-    accounts = queries.get_aws_accounts(terraform_state=True, ecrs=False)
+        # on the tgw definition in the cluster file.
+        else {}
+    )
 
-    # Fetch desired state for cluster-to-vpc(account) VPCs
-    with AWSApi(1, accounts, settings=settings, init_users=False) as awsapi:
-        desired_state, err = build_desired_state_tgw_attachments(
-            clusters, ocm_map, awsapi
-        )
-    if err:
-        sys.exit(1)
 
-    # check there are no repeated vpc connection names
+def _validate_vpc_connection_names(desired_state: list) -> None:
     connection_names = [c["connection_name"] for c in desired_state]
     if len(set(connection_names)) != len(connection_names):
         logging.error("duplicate vpc connection names found")
         sys.exit(1)
```

Comment:
Since this is a full refactor: lets also `raise` here.

Reply:
updated